ballerina-to-openapi/src/main/java/io/ballerina/openapi/service/mapper/response/utils/StatusCodeTypeUtils.java

```diff
@@ -0,0 +1,118 @@
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package io.ballerina.openapi.service.mapper.response.utils;
+
+import io.ballerina.compiler.api.SemanticModel;
+import io.ballerina.compiler.api.symbols.RecordFieldSymbol;
+import io.ballerina.compiler.api.symbols.RecordTypeSymbol;
+import io.ballerina.compiler.api.symbols.Symbol;
+import io.ballerina.compiler.api.symbols.TypeDefinitionSymbol;
+import io.ballerina.compiler.api.symbols.TypeReferenceTypeSymbol;
+import io.ballerina.compiler.api.symbols.TypeSymbol;
+import io.ballerina.openapi.service.mapper.response.model.HeaderRecordInfo;
+import io.ballerina.openapi.service.mapper.type.TypeMapper;
+import io.ballerina.openapi.service.mapper.utils.MapperCommonUtils;
+import io.swagger.v3.oas.models.headers.Header;
+import io.swagger.v3.oas.models.media.Schema;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * This {@link StatusCodeTypeUtils} class provides common functionalities for mapping the Ballerina HTTP status code
+ * response and error to OpenAPI response.
+ *
+ * @since 1.9.0
+ */
+public abstract class StatusCodeTypeUtils {
+
+     static boolean isSubTypeOfBallerinaModuleType(String type, String moduleName, TypeSymbol typeSymbol,
+                                                   SemanticModel semanticModel) {
+        Optional<Symbol> optionalRecordSymbol = semanticModel.types().getTypeByName("ballerina", moduleName,
+                "", type);
+        if (optionalRecordSymbol.isPresent() &&
+                optionalRecordSymbol.get() instanceof TypeDefinitionSymbol recordSymbol) {
+            return typeSymbol.subtypeOf(recordSymbol.typeDescriptor());
+        }
+        return false;
+    }
+
+    static RecordTypeSymbol getRecordTypeSymbol(TypeSymbol typeSymbol, TypeMapper typeMapper) {
+        TypeSymbol statusCodeResType = typeMapper.getReferredType(typeSymbol);
+        RecordTypeSymbol statusCodeRecordType = null;
+        if (statusCodeResType instanceof TypeReferenceTypeSymbol statusCodeResRefType &&
+                statusCodeResRefType.typeDescriptor() instanceof RecordTypeSymbol recordTypeSymbol) {
+            statusCodeRecordType = recordTypeSymbol;
+        } else if (statusCodeResType instanceof RecordTypeSymbol recordTypeSymbol) {
+            statusCodeRecordType = recordTypeSymbol;
+        }
+        return statusCodeRecordType;
```

Comment:
can't we follow the same format and return an optional here as it can be null as well

Reply:
Fixed via [4ae8d86](https://github.com/ballerina-platform/openapi-tools/pull/1604/commits/4ae8d8652f0245bcf25ebb15db8f268844a02fa8)